circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -389,3 +371,80 @@ def _get_pauli_indices(cog: CommutingObservableGroup) -> list[int]:
     if not pauli_indices:
         pauli_indices = [0]
     return pauli_indices
+
+
+def _consolidate_resets(circuit: QuantumCircuit, inplace=True) -> QuantumCircuit:
+    """Consolidate redundant resets into a single reset."""
+    if not inplace:  # pragma: no cover
+        circuit = deepcopy(circuit)
```

Comment:
why not just `qc.copy()`? which is what we have done in other places in case of `if not inplace`